Mathlib/Analysis/Convex/Radon.lean

```diff
@@ -0,0 +1,232 @@
+convex hulls of `I` and `Iᶜ` intersect. -/
+theorem radon_partition {ι : Type*} {f : ι → E}
+    (h : ¬AffineIndependent 𝕜 f) : ∃ (I : Set ι),
+    (Set.Nonempty ((convexHull 𝕜 (f '' I)) ∩ (convexHull 𝕜 (f '' Iᶜ)))) := by
+  unfold AffineIndependent at h; push_neg at h
+  rcases h with ⟨s1, w, h_wsum, h_vsum, nonzero_w_index, h1, h2⟩
+  let I : Finset ι := s1.filter (fun i => w i ≥ 0)
+  let J : Finset ι := s1.filter (fun i => w i < 0)
+  use (I : Set ι)
+
+  let weights_sum_I := ∑ i in I, w i
+  let weights_sum_J := ∑ i in J, w i
+  have h3 : weights_sum_I + weights_sum_J = 0 := by
+    rw [←h_wsum]; simp
+    have h3_aux := Finset.sum_filter_add_sum_filter_not s1 (fun i => 0 ≤ w i) w
+    simp at h3_aux; exact h3_aux
+
+  have h4 : (weights_sum_I > 0) ∧ (weights_sum_J < 0) := by
+    rcases (lt_or_gt_of_ne h2) with h_neg | h_pos
+    have h_J_neg : weights_sum_J < 0 := by
+      rw [←neg_pos, ←Finset.sum_neg_distrib]
+      let neg_w : ι → 𝕜 := fun i => -w i
+      have h4_aux : neg_w nonzero_w_index ≤ (Finset.sum J neg_w) := by
+        apply Finset.single_le_sum
+        intros i hi
+        simp; rw [Finset.mem_filter] at hi; linarith
+        simp; constructor; exact h1; exact h_neg
+      simp at h4_aux; simp; linarith
+    constructor
+    linarith
+    exact h_J_neg
+
+    have h_I_pos : weights_sum_I > 0 := by
+      have h4_aux : weights_sum_I ≥ w nonzero_w_index := by
+        apply Finset.single_le_sum
+        intros i hi
+        rw [Finset.mem_filter] at hi; exact hi.right
+        simp; constructor; exact h1; exact LT.lt.le h_pos
+      linarith
+    constructor
+    exact h_I_pos
+    linarith
+  rcases h4 with ⟨h_I_pos, h_J_neg⟩
+
+  let w' : ι → 𝕜 := fun i => if w i ≥ 0
+    then (w i) / weights_sum_I
+    else -(w i) / weights_sum_I
+  let p : E := ∑ i in I, (w' i) • (f i) -- point of intersection
+
+  have h5_I : ∑ i in I, w' i = 1 := by
+    let w'' : ι → 𝕜 := fun i => (w i) / weights_sum_I
+    have h5_aux : ∑ i in I, w' i = ∑ i in I, w'' i := by
+      apply Finset.sum_congr; rfl
+      intros i hi
+      rw [Finset.mem_filter] at hi
+      rcases hi with ⟨_, hh⟩
+      simp
+      contrapose; intros
+      exact Iff.mpr not_lt hh
+    rw [h5_aux]
+    simp
+    rw [←Finset.sum_div]
+    apply div_self
+    exact ne_of_gt h_I_pos
+
+  have h5_J : ∑ i in J, w' i = 1 := by
+    let w'' : ι → 𝕜 := fun i => (w i) / weights_sum_J
+    have h5_aux1 : weights_sum_I = -weights_sum_J := by
+      exact Iff.mp add_eq_zero_iff_eq_neg h3
+    have h5_aux2 : ∑ i in J, w' i = ∑ i in J, w'' i := by
+      apply Finset.sum_congr; rfl
+      intros i hi
+      rw [Finset.mem_filter] at hi
+      rcases hi with ⟨_, h_neg⟩
+      simp
+      split_ifs
+      exfalso; linarith
+      simp_rw [h5_aux1, div_neg]
+      ring
+    rw [h5_aux2]; simp
+    rw [←Finset.sum_div]
+    apply div_self
+    exact ne_of_lt h_J_neg
+
+  have h6 : ∀ v, w' v ≥ 0 := by
+    intro; simp; split_ifs
+    repeat {apply div_nonneg; linarith; linarith}
```

Comment:
```suggestion
  have h6 : ∀ v, 0 ≤ w' v := by
    intro; exact div_nonneg (abs_nonneg _) h_I_pos.le
```
Reversing the inequalities as usual, and here the proof becomes simpler thanks to the new definition of `w'`